test.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+set -e
+
+cd ~
+
+# Enter a virtualenv we control!
+python -m virtualenv .
+source bin/activate
+pip install --upgrade pip
+
+# install ros and some dependencies
+sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+sudo apt-key adv --keyserver hkp://pgp.mit.edu:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+sudo apt-get update -qq
+sudo apt-get install -y python-catkin-pkg python-rosdep ros-indigo-catkin ros-indigo-ros ros-indigo-roslaunch build-essential ros-indigo-rosnode
+
+# Install our project
+pip install -e ~/$CIRCLE_PROJECT_REPONAME
+
+# And copy it into the catkin_ws directory
+cp -r ~/pyrostest/test/pyrostest ~/catkin_ws/src
```

Comment:
Two entirely unrelated things are happening. The first line installs the python package (so that `import pyrostest` works), and the second line moves the Ros package into the catkins_ws dir. You can't pip install to a destination because pip installs are always system (or venv) wide. They aren't installed anywhere (or at least anywhere that matters).